workers/price-reporter/src/mock.rs

```diff
@@ -0,0 +1,219 @@
+    /// The price to report for all pairs
+    price: Price,
+    /// The queue on which to accept jobs
+    job_queue: PriceReporterReceiver,
+}
+
+impl MockPriceReporter {
+    /// Create a new mock price reporter
+    pub fn new(price: Price, job_queue: PriceReporterReceiver) -> Self {
+        Self { price, job_queue }
+    }
+
+    /// Start the mock price reporter
+    pub fn run(self) {
+        thread::spawn(move || {
+            let rt = TokioRuntime::new().unwrap();
+            rt.block_on(self.execution_loop());
+        });
+    }
+
+    /// The execution loop of the mock price reporter
+    async fn execution_loop(mut self) {
+        loop {
+            let job = self.job_queue.recv().await;
+            if let Some(job) = job {
+                if let Err(e) = self.handle_job(job).await {
+                    log::error!("error in mock price reporter: {e:?}");
+                }
+            }
+        }
+    }
+
+    /// Handle a job
+    async fn handle_job(&self, job: PriceReporterJob) -> Result<(), PriceReporterError> {
+        match job {
+            PriceReporterJob::StartPriceReporter { .. } => {
+                log::debug!("mock price reporter got `StartPriceReporter` job");
+                Ok(())
+            },
+            PriceReporterJob::PeekMedian { base_token, quote_token, channel } => {
+                self.handle_peek_median(base_token, quote_token, channel).await
+            },
+            PriceReporterJob::PeekAllExchanges { base_token, quote_token, channel } => {
+                self.handle_peek_all_exchanges(base_token, quote_token, channel).await
+            },
+        }
+    }
+
+    /// Handle a peek median job
+    async fn handle_peek_median(
+        &self,
+        base_token: Token,
+        quote_token: Token,
+        channel: OneshotSender<PriceReporterState>,
+    ) -> Result<(), PriceReporterError> {
+        // Construct a state and send it back on the queue
+        let timestamp = get_current_time_seconds();
+        let state = PriceReporterState::Nominal(PriceReport {
+            base_token,
+            quote_token,
+            exchange: None, // midpoint
+            midpoint_price: self.price,
+            local_timestamp: timestamp,
+            reported_timestamp: Some(timestamp as u128),
+        });
+
+        if let Err(e) = channel.send(state) {
+            log::error!("error sending price report: {e:?}");
+        }
+
+        Ok(())
+    }
+
+    /// Handle a peek all exchanges job
+    async fn handle_peek_all_exchanges(
+        &self,
+        base_token: Token,
+        quote_token: Token,
+        channel: OneshotSender<HashMap<Exchange, ExchangeConnectionState>>,
+    ) -> Result<(), PriceReporterError> {
+        // Construct a state and send it back on the queue
+        let timestamp = get_current_time_seconds();
+        let report = PriceReport {
+            base_token,
+            quote_token,
+            exchange: Some(Exchange::Binance),
```

Comment:
does it matter that this is the exchange for all of the `ExchangeConnectionState`s?

Reply:
It doesn't matter but I will change it so it reads better, nice catch